packages/shared/src/authz/policies/canQueryProject.ts

```diff
@@ -18,44 +16,45 @@ import {
   WorkspaceNoAccessError,
   WorkspaceSsoSessionInvalidError
 } from '../domain/authErrors.js'
+import { err, isOk, ok } from 'true-myth/result'
+import { AuthCheckContextLoaderKeys } from '../domain/loaders.js'
+import { LogicError } from '../domain/errors.js'
 
-export const canQueryProjectPolicyFactory =
-  (
-    loaders: Pick<
-      AuthCheckContextLoaders,
-      | 'getEnv'
-      | 'getProject'
-      | 'getProjectRole'
-      | 'getServerRole'
-      | 'getWorkspaceRole'
-      | 'getWorkspaceSsoProvider'
-      | 'getWorkspaceSsoSession'
-    >
-  ) =>
-  async ({
-    userId,
-    projectId
-  }: UserContext & ProjectContext): Promise<
-    AuthResult<
-      | typeof ProjectNotFoundError
-      | typeof ProjectNoAccessError
-      | typeof WorkspaceNoAccessError
-      | typeof WorkspaceSsoSessionInvalidError
-    >
-  > => {
-    const { FF_ADMIN_OVERRIDE_ENABLED, FF_WORKSPACES_MODULE_ENABLED } = loaders.getEnv()
+export const canQueryProjectPolicyFactory: AuthPolicyFactory<
```

Comment:
Instead of all of the arg, loader & return types being typed out next to the actual args, loaders & return, I instead created an AuthPolicyFactory type that the policy factory constant is typed at.

This ensures that the policy factory follows an expected schema, that it can't randomly decided to have more positional args next to loaders, or return a non-AuthResult etc.

All of the loader, arg & return type requirements are also defined here as generic args of AuthPolicyFactory